java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -24,12 +32,54 @@ import arcs.core.entity.ReadWriteSingletonHandle
 import arcs.core.entity.Reference
 import arcs.core.entity.WriteCollectionHandle
 import arcs.core.entity.WriteSingletonHandle
-import arcs.core.storage.Handle as StorageHandle
+import arcs.core.storage.ActivationFactory
+import arcs.core.storage.Dereferencer
+import arcs.core.storage.Reference as StorageReference
 import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageMode.ReferenceMode
+import arcs.core.storage.StorageProxy
 import arcs.core.storage.handle.CollectionHandle
+import arcs.core.storage.handle.CollectionProxy
+import arcs.core.storage.handle.CollectionStoreOptions
 import arcs.core.storage.handle.HandleManager
+import arcs.core.storage.handle.RawEntityDereferencer
 import arcs.core.storage.handle.SingletonHandle
+import arcs.core.storage.handle.SingletonProxy
+import arcs.core.storage.handle.SingletonStoreOptions
+import arcs.core.storage.handle.Stores
 import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.Time
+
+class EntityPrep<T : Entity>(
+    val handleName: String,
+    val idGenerator: Id.Generator,
+    val schema: Schema,
+    val ttl: Ttl,
+    val time: Time
+) {
+    fun prepareEntity(entity: T): RawEntity {
+        entity.ensureIdentified(idGenerator, handleName)
+
+        val rawEntity = entity.serialize()
+
+        @Suppress("GoodTime") // use Instant
+        rawEntity.creationTimestamp = requireNotNull(time).currentTimeMillis
+        require(schema.refinement(rawEntity)) {
+            "Invalid entity stored to handle $handleName(failed refinement)"
+        }
+        if (!Ttl.Infinite.equals(ttl)) {
```

Comment:
IntelliJ keeps suggesting this to me. I keep changing it, but never end up submitting it...

```
if (ttl != Ttl.Infinite) {
```